.github/workflows/test.yml

```diff
@@ -0,0 +1,51 @@
+name: Test
+
+# Controls when the action will run.
+# Triggers the workflow on push or pull request events but only for the master branch
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-16.04, ubuntu-latest, windows-latest, macos-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Cache node modules
+      id: npm-cache
+      uses: actions/cache@v1
+      env:
+        cache-name: test
+      with:
+        path: node_modules
+        key:  ${{ runner.os }}-npm-cache-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-npm-cache-${{ env.cache-name }}-
+          ${{ runner.os }}-npm-cache-
+
+    - name: Install dependencies
+      if: steps.npm-cache.outputs.cache-hit != 'true'
+      run: npm install --no-audit --no-fund
+    
+    - name: Check type errors
+      run: npx tsc --noEmit
+
+    - name: Run tests
+      if: matrix.os != 'ubuntu-16.04'
+      run: npm test -- --coverage
```

Comment:
Do we need to generate a coverage report for non-16.04 pipelines?

Reply:
No, it seems to just upload the same thing.

I think I could easily get it to do code coverage for the other operating systems, but doing a coverage test is slower, and I only care about non-16.04 for making sure devs don't have trouble.

In theory there could be something that only happens in certain operating systems. I could try doing coverage for all if you want.

Reply:
Oh I see what you meant now. That's a typo. Will fix.